diff -purN a/client/dhclient.c b/client/dhclient.c
--- a/client/dhclient.c	2016-07-20 16:31:22.000000000 -0600
+++ b/client/dhclient.c	2016-07-22 10:49:18.180826576 -0600
@@ -344,7 +344,7 @@ main(int argc, char **argv) {
 		} else if (!strcmp(argv[i], "-v")) {
 			quiet = 0;
 		} else if (!strcmp(argv[i], "--version")) {
-			log_info("isc-dhclient-%s", PACKAGE_VERSION);
+			log_info("isc-dhclient-%s-jaguar", PACKAGE_VERSION);
 			exit(0);
 		} else if (!strcmp(argv[i], "-I")) {
 			if ((++i == argc) || (argv[i] == NULL) || (*(argv[i])=='\0')) {
diff -purN a/relay/dhcrelay.c b/relay/dhcrelay.c
--- a/relay/dhcrelay.c	2016-07-20 16:31:22.000000000 -0600
+++ b/relay/dhcrelay.c	2016-07-22 10:49:58.164959411 -0600
@@ -383,7 +383,7 @@ main(int argc, char **argv) {
 		} else if (!strcmp(argv[i], "--no-pid")) {
 			no_pid_file = ISC_TRUE;
 		} else if (!strcmp(argv[i], "--version")) {
-			log_info("isc-dhcrelay-%s", PACKAGE_VERSION);
+			log_info("isc-dhcrelay-%s-jaguar", PACKAGE_VERSION);
 			exit(0);
 		} else if (!strcmp(argv[i], "--help") ||
 			   !strcmp(argv[i], "-h")) {
diff -purN a/server/dhcp.c b/server/dhcp.c
--- a/server/dhcp.c	2016-07-20 16:31:22.000000000 -0600
+++ b/server/dhcp.c	2016-07-25 09:42:40.782749330 -0600
@@ -36,6 +36,10 @@
 #include <errno.h>
 #include <limits.h>
 #include <sys/time.h>
+#include <stdio.h> /* JAGUAR */
+#include <unistd.h> /* JAGUAR */
+#include <sys/wait.h> /* JAGUAR */
+#include <sys/types.h> /* JAGUAR */
 #include "trace.h"
 static void commit_leases_ackout(void *foo);
 static void maybe_return_agent_options(struct packet *packet,
@@ -116,6 +120,52 @@ char *print_hw_addr_or_client_id (packet
 				      packet -> raw -> chaddr);
 }
 
+static void log_lease_exhaustion(packet) /* JAGUAR */
+	struct packet *packet;
+{
+	char * name;
+	u_int8_t hlen;
+	u_int8_t * hbuf;
+	pid_t pid;
+	char * argv[5];
+	char buffer[(HARDWARE_ADDR_LEN + 1) * 3];
+	char * here;
+	int status;
+
+	if (!packet || !(packet -> shared_network) || !(packet -> shared_network -> name) || !(packet -> haddr)) { return; }
+
+	name = packet -> shared_network -> name;
+	hlen = packet -> haddr -> hlen;
+	hbuf = packet -> haddr -> hbuf;
+
+	if (hlen > (HARDWARE_ADDR_LEN + 1)) { hlen = HARDWARE_ADDR_LEN + 1; }
+	if (hlen > 0) { --hlen; ++hbuf; /* Skip type byte on MAC address. */ }
+
+	for (here = buffer; hlen > 0; --hlen) {
+		here += snprintf(here, 3, "%02x", *(hbuf++));
+		if (hlen > 1) { *(here++) = ':'; }
+	}
+
+	argv[0] = "/usr/local/sbin/dhcpindicate";
+	argv[1] = "failure";
+	argv[2] = name;
+	argv[3] = buffer;
+	argv[4] = NULL;
+
+	if ((pid = fork()) > 0) {
+		waitpid(pid, &status, 0);
+		if (status) {
+			log_error("log_lease_exhaustion: %s exit status %d", argv[0], status);
+		}
+	} else if (pid == 0) {
+		execvp(argv[0], argv);
+		log_error("log_lease_exhaustion: Unable to execute %s: %m", argv[0]);
+		_exit(127);
+	} else {
+		log_error("log_lease_exhaustion: fork() failed");
+	}
+}
+
 void
 dhcp (struct packet *packet) {
 	int ms_nulltp = 0;
@@ -298,6 +348,10 @@ dhcp (struct packet *packet) {
       out:
 	if (lease)
 		lease_dereference (&lease, MDL);
+
+#if 0
+	log_lease_exhaustion(packet); /* JAGUAR UNIT TEST */
+#endif
 }
 
 void dhcpdiscover (packet, ms_nulltp)
@@ -388,10 +442,12 @@ void dhcpdiscover (packet, ms_nulltp)
 			if (peer_has_leases)
 				log_error ("%s: peer holds all free leases",
 					   msgbuf);
-			else
+			else {
 				log_error ("%s: network %s: no free leases",
 					   msgbuf,
 					   packet -> shared_network -> name);
+				log_lease_exhaustion(packet); /* JAGUAR */
+			}
 			return;
 		}
 	}
diff -purN a/server/dhcpd.c b/server/dhcpd.c
--- a/server/dhcpd.c	2016-07-20 16:31:22.000000000 -0600
+++ b/server/dhcpd.c	2016-07-22 10:50:04.468295644 -0600
@@ -399,7 +399,7 @@ main(int argc, char **argv) {
 			local_family_set = 1;
 #endif /* DHCPv6 */
 		} else if (!strcmp (argv [i], "--version")) {
-			log_info("isc-dhcpd-%s", PACKAGE_VERSION);
+			log_info("isc-dhcpd-%s-jaguar", PACKAGE_VERSION);
 			exit (0);
 #if defined (TRACING)
 		} else if (!strcmp (argv [i], "-tf")) {
